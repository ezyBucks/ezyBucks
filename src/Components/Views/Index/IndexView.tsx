import * as React from 'react';

/**
 * Class to display the main landing page when not logged in.
 */
export const IndexView: React.SFC = () => {
    return (
        <div>
            <h1>ezyBuccs</h1>
        </div>
    );
};
