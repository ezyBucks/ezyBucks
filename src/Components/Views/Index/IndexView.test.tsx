import React from 'react';
import renderer from 'react-test-renderer';
import { IndexView } from './IndexView';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

it('renders without crashing', () => {
    shallow(<IndexView />);
});

it('fully renders without crashing', () => {
    const wrapper = mount(
        <MemoryRouter>
            <IndexView />
        </MemoryRouter>
    );
    expect(wrapper.contains(<IndexView />)).toBe(true);
});

it('matches the snapshot', () => {
    const index = renderer.create(<IndexView />).toJSON();
    expect(index).toMatchSnapshot();
});
