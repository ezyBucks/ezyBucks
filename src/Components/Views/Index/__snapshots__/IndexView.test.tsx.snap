// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div
    className="css-337oej"
  >
    <div
      className="css-dvxtzn"
      style={
        Object {
          "marginTop": "30px",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "width": "25%",
          }
        }
      >
        <span
          className="ant-avatar ant-avatar-square"
          style={
            Object {
              "backgroundColor": "#1890ff",
              "float": "left",
              "fontSize": "25px",
              "height": 64,
              "lineHeight": "64px",
              "width": 64,
            }
          }
        >
          <span
            className="ant-avatar-string"
          >
            <b>
              RC
            </b>
          </span>
        </span>
        <span>
          <b
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            Robert Calvert
          </b>
          <br />
          rcalvert@live.com
        </span>
        <div
          style={
            Object {
              "marginTop": "50px",
            }
          }
        >
          <div
            className="css-dvxtzn"
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            <b>
              Account Balance
            </b>
            <br />
            2786
             ezy bucks
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
